test/src/e2e_vm_tests/test_programs/should_pass/language/main_args/main_args_empty/test.encoding_v1.toml

```diff
@@ -0,0 +1,4 @@
+script_data = "0000000000000538 0000000000000001"
+expected_result = { action = "return_data", value = "0000000000000539" }
+
+experimental = { new_encoding = true }
```

Comment:
This is actually a proposal for the file name. Right now it is `test.encoding_v1.toml`. As I understood, we can name the file arbitrarily, and also enable arbitrary experimental features in it which is great and offers perfect flexibility.

But for the most regular case where a single experimental feature will have its own `test.<feature>.toml`, for the sake of consistency I would propose to always use the experimental feature name in the file name. This will save unnecessary guessing.

It this case, it would mean having `test.new_encoding.toml`. (Or we rename the feature flag to `encoding_v1` which would actually be my preference, `new` is somehow too generic.)

Reply:
I agree. I will rename it.

> It this case, it would mean having test.new_encoding.toml. (Or we rename the feature flag to encoding_v1 which would actually be my preference, new is somehow too generic.)

I agree. This is named `encoding_v1` because I forgot to rename it. The issue was that I started with feature named `encoding_v1`, but this demanded changing `sway-std-lib` conditional compilations flags.

And this caused the issue of the current version of the compiler would not compile older `sway-std-lib` anymore.

I think this is too much just for the renaming.

Reply:
Just to ask here, since you've mentioned you will rename the file, did you forget or rather decided in the end to keep the current name :smile: